jina/types/arrays/memmap.py

```diff
@@ -150,39 +163,121 @@ def append(self, doc: 'Document', flush: bool = True) -> None:
                 ],
             ).tobytes()
         )
-        self._header_map[doc.id] = (len(self._header_map), p, r, r + l)
+        if idx is None:
+            # the idx of the appended document is the idx of the last doc + 1
+            idx = (
+                next(reversed(self._header_map.items()))[1][0] + 1
+                if len(self._header_map)
+                else 0
+            )
+            self._header_map[doc.id] = (idx, p, r, r + l)
+            self._header_map.move_to_end(doc.id)
+        else:
+            self._header_map.update({doc.id: (idx, p, r, r + l)})
```

Comment:
Not sure about it, but does the `update` performs equally well as the pure assignment? In the end, the update here needs to first create a new dictionary in memory with one entry and then assign. This seems to be a performance loss.

Reply:
I've changed it